webserver/views/api/v1/core.py

```diff
@@ -267,6 +289,97 @@ def get_many_highlevel():
     return jsonify(recording_details)
 
 
+def parse_select_features():
+    """Check whether the features are found or not.
+    Parse the query string of features to create a list of 
+    the json paths to features, excluding those that are 
+    not offered.
+
+    If features are missing, an APIBadRequest Exception is
+    raised stating the missing features message.
+    
+    Returns:
+        parsed_features, a list of tuples of the form:
+            (<feature_path>, <alias>, <default_type>)
+            
+            <feature_path> is a string holding the path to a feature:
+            "llj.data->feature_name"
+
+            <alias> is a string alias for a feature:
+            "lowlevel.feature_name"
+
+            <default_type> is the type to which a feature value will default,
+            if it is non-existent, depending on the alias from AVAILABLE_FEATURES.
+    """
+    features_param = request.args.get("features")
+    if not features_param:
+        raise webserver.views.api.exceptions.APIBadRequest("Missing `features` parameter")
+
+    parsed_features = []
+    for alias in features_param.split(';'):
+        if alias in AVAILABLE_FEATURES:
+            default_type = AVAILABLE_FEATURES[alias][1]
+            feature_path = AVAILABLE_FEATURES[alias][0]
+            parsed_features.append((feature_path, alias, default_type))
+
+    # Always include metadata.version and metadata.audio_properties
+    metadata_version = "llj.data->'metadata'->'version'"
+    metadata_version_alias = "metadata.version"
+    metadata_audio_properties = "llj.data->'metadata'->'audio_properties'"
+    metadata_audio_properties_alias = "metadata.audio_properties"
+    
+    parsed_features.append((metadata_version, metadata_version_alias, {}))
+    parsed_features.append((metadata_audio_properties, metadata_audio_properties_alias, {}))
+
+    # Remove duplicates, preserving order
+    ret = []
+    return [x for x in parsed_features if not (x in ret or ret.append(x))]
```

Comment:
@alastair although we added the remove duplicates utility, that method did not work once I added support for default types - some of the default types were {}, and they are unhashable so it didn't work with the usage of set(). Is this a suitable method instead?

Reply:
OK, if you need to use the same functionality but in a different form then that's fine to duplicate it